inveniordm_py/records/metadata.py

```diff
@@ -39,3 +40,37 @@ class RecordListMetadata(ListMetadata):
 
     accept = "application/vnd.inveniordm.v1+json"
     item_class = RecordMetadata
+
+
+class RecordCommunityMetadata(Metadata):
+    """Record community metadata class."""
+
+    accept = "application/json"
+    content_type = "application/json"
+
+    def _serialize_data(self):
+        """Serialize community data to InvenioRDM format."""
+        _data = self._data.get("communities", [])
+        if not _data:
+            return {}
+
+        req = {"communities": []}
+        if isinstance(_data, list):
+            req["communities"] = [{"id": id_} for id_ in _data]
+        elif isinstance(_data, str):
+            req["communities"] = [{"id": _data}]
+        else:
+            raise ValueError(
+                f"Invalid data type for community metadata. Expected: list or str. Got: {type(_data)}"
+            )
+        return req
+
+    def to_request(self):
+        """Convert metadata to request body."""
+        return json.dumps(self._serialize_data())
+
+
+class RecordCommunitiesListMetadata(ListMetadata):
+    """Record communities list metadata class."""
+
+    item_class = RecordCommunityMetadata
```

Comment:
Comment: Having a `RecordCommunityMetadata` class that supports a list of communities looks a bit redundant with having `RecordCommunitiesListMetadata` wich supports a list of `RecordCommunityMetadata`. Could you explaing the use case of why we have both?

Reply:
In the client's design, `ListMetadata` objects support `search` methods (e.g. responses from the API that are generated from ES/OS). I.e., it implements `hits` and `aggregations` properties. Then, each object is a `RecordCommunityMetadata`. 

On the other hand, `Metadata` objects implement more generic representations of metadata. I.e. `RecordCommunityMetadata` refers to **one** record community.